packages/astro/src/runtime/client/dev-toolbar/apps/astro.ts

```diff
@@ -372,8 +373,11 @@ export default {
 					resetDebugButton();
 				}, 3500);
 			});
-
 			canvas.append(windowComponent);
+
+			// If we have integration data, rebuild that part of the UI as well
+			// as it probably mean that the user had already open the app in this session (ex: view transitions)
+			if (integrationData) refreshIntegrationList();
```

Comment:
Moved this here because previously it'd refetch on every page load when using view transitions, we only want to fetch this once.